src/core/friendly_errors/sketch_reader.js

```diff
@@ -0,0 +1,396 @@
+
+      start = code.indexOf('/*');
+      end = code.indexOf('*/');
+    }
+
+    return code;
+  };
+
+  /**
+   * Checks if a p5.js constant or a function is
+   * declared outside a function and reports it if found.
+   *
+   * @method globalConstFuncCheck
+   * @private
+   * @returns {Boolean}
+   */
+
+  const globalConstFuncCheck = () => {
+    // generate all the const key data as an array
+    const tempArray = Object.keys(constants);
+    let element;
+    let isFound = false;
+    for (let i = 0; i < tempArray.length; i++) {
+      try {
+        //if the user has not declared p5.js constant anywhere outside the
+        //setup or draw function then this will throw an
+        //error.
+        element = eval(tempArray[i]);
+      } catch (e) {
+        //We are catching the error due to the above mentioned
+        //reason. Since there is no declaration of constant everything
+        //is OK so we will skip the current iteration and check for the
+        //next element.
+        continue;
+      }
+      //if we are not getting an error this means
+      //user have changed the value. We will check
+      //if the value is changed and if it is changed
+      //then report.
+      if (constants[tempArray[i]] !== element) {
+        let url = `https://p5js.org/reference/#/p5/${tempArray[i]}`;
+        p5._friendlyError(
+          translator('fes.sketchReaderErrors.reservedConst', {
+            url: url,
+            symbol: tempArray[i]
+          })
+        );
+        isFound = true;
+        break;
+      }
+    }
+    //if a p5.js constant is already reported then no need to check
+    //for p5.js functions.
+    if (!isFound) {
+      //the below code gets a list of p5.js functions
+      let p5Constructors = {};
+      for (let key of Object.keys(p5)) {
+        // Get a list of all constructors in p5. They are functions whose names
+        // start with a capital letter
+        if (typeof p5[key] === 'function' && key[0] !== key[0].toLowerCase()) {
+          p5Constructors[key] = p5[key];
+        }
+      }
+      const keyArray = Object.keys(p5Constructors);
+      let functionArray = [];
+      //get the names of all p5.js functions
+      for (let i = 0; i < keyArray.length; i++) {
+        functionArray.push(
+          ...Object.keys(p5Constructors[keyArray[i]].prototype)
+        );
+      }
+      functionArray = functionArray.filter(ele => !ele.includes('_'));
+
+      //we have p5.js function names with us so we will check
+      //if they have been declared or not.
+      for (let i = 0; i < functionArray.length; i++) {
+        //ignoreFunction contains the list of functions to be ignored
+        if (!ignoreFunction.includes(functionArray[i])) {
+          try {
+            //if we get an error that means the function is not declared
+            element = eval(functionArray[i]);
+          } catch (e) {
+            //we will skip the iteration
+            continue;
+          }
```

Comment:
I don't see the index `i` being used anywhere else, maybe you can use a for-of instead:
```javascript
for (let function_name of functionArray) { ... }
```

Reply:
I used for loop here because when p5.js is bundled it is converted to ES5 format and this conversion sometimes causes problems especially with the new variations of for loop. So to be on the safer side I used the traditional for loop.

Reply:
I see, that makes sense. I'll mark this as resolved then